### ISSUE TITLE
Recurrent Neural Network update on sequence of length 1

### ISSUE THREAD

**Issue by HarrieO** [Element-Research/rnn#7]
The Recurrent class of 'rnn' does not allow training updates on sequences of length 1. Minimal example:

      require 'rnn'

      x = torch.rand(200)
      target = torch.rand(1)

      rho = 5
      hiddenSize = 100
      -- RNN
      r = nn.Recurrent(
         hiddenSize, nn.Linear(200,hiddenSize), 
         nn.Linear(hiddenSize, hiddenSize), nn.Sigmoid(), 
         rho
      )

      seq = nn.Sequential()
      seq:add(r)
      seq:add(nn.Linear(hiddenSize, 1))

      criterion = nn.MSECriterion()

      output = seq:forward(x)
      err = criterion:forward(output,target)
      gradOutput = criterion:backward(output,target)
      seq:backward(x,gradOutput)

      seq:updateParameters(0.01)

As far as I understand this should not be an issue, yet when ran this gives something like:

      /Users/hroosterhuis/torch/install/bin/luajit: /Users/hroosterhuis/torch/install/share/lua/5.1/nn/Add.lua:62: bad argument #1 to 'size' (dimension 1 out of range of 0D tensor at /Users/hroosterhuis/torch/pkg/torch/generic/Tensor.c:17)
      stack traceback:
      [C]: in function 'size'
      /Users/hroosterhuis/torch/install/share/lua/5.1/nn/Add.lua:62: in function 'accGradParameters'
      ...s/hroosterhuis/torch/install/share/lua/5.1/nn/Module.lua:53: in function 'accUpdateGradParameters'
      ...oosterhuis/torch/install/share/lua/5.1/rnn/Recurrent.lua:247: in function 'accUpdateGradParametersThroughTime'
      ...is/torch/install/share/lua/5.1/rnn/AbstractRecurrent.lua:73: in function 'backwardUpdateThroughTime'
      ...is/torch/install/share/lua/5.1/rnn/AbstractRecurrent.lua:83: in function 'updateParameters'
      ...roosterhuis/torch/install/share/lua/5.1/nn/Container.lua:31: in function 'updateParameters'
      testRNN.lua:26: in main chunk
      [C]: in function 'dofile'
      ...huis/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
      [C]: at 0x0103773780

**Comment by nicholas-leonard** [comment]
Ok. Should be fixed now. git pull and luarocks make. Let me know.

**Comment by HarrieO** [comment]
I just checked and it seems to work fine now.
Thanks a lot!
